src/v2/port_mapping.in.rs

```diff
@@ -145,25 +185,40 @@ impl fmt::Display for PortMapping {
         if let Some(ports) = self.host_ports {
             write!(f, "{}:", ports)?;
         }
-        write!(f, "{}", self.container_ports)
+        write!(f, "{}/", self.container_ports)?;
+        write!(f, "{}", self.protocol)
     }
 }
 
+fn consume_protocol(ports_and_protocol: &str) -> Result<(&str, Protocol)> {
+    let fields: Vec<_> = ports_and_protocol.split("/").collect();
+    match fields.len() {
+        1 => Ok((fields[0], Protocol::TCP)),
+        2 => Ok((fields[0], FromStr::from_str(fields[1])?)),
```

Comment:
You can write `Protocol::from_str` in Rust (using the name of the class, and not just the trait), and it would probably be clearer in this case.

Reply:
Agreed. I copied this from a little further down where it does:
```
//...
                    host_ports: Some(FromStr::from_str(fields[1])?),
                    container_ports: FromStr::from_str(fields[0])?,
//...
```